tutorials/ui/gui_theme_type_variations.rst

```diff
@@ -0,0 +1,50 @@
+.. _doc_gui_theme_type_variations:
+
+Overview
+========
+
+When designing a user interface there may be times when a control node
+needs to have a different look than what is normally defined by the
+theme. Theme override properties of a node can be used. However if
+this change needs to be done multiple times the same work is being done
+multiple times, it also requires finding and editing every individual
+node with that look if it needs to be adjusted.
+
+As a solution to this problem there are theme type variations. These
+are used to create variations of control nodes in a way that is part
+of the theme.
+
+These can be used to redefine every aspect of a control node or only one.
+For example, if a button has a defined StyleBoxFlat for its normal state in
+the theme, and a button type variation only has a different font specified,
+any button using that type variation will still use the StyleBoxFlat defined
+for the theme.
```

Comment:
```suggestion
Type variations can replace some aspects of the base type, but keep others.
They can also define properties that the base style hasn't defined. For example,
your ``GrayButton`` can override the ``normal`` style from the base ``Button``
and add ``font_color`` that ``Button`` has never defined. The control will use
a combination of both types giving priority to the type variation.

.. note::
   The way controls resolve what theme items they use from each type and each
   theme is better described in the "Customizing a control" section of :ref:`doc_gui_skinning`.
```


Might also be worth adding an anchor to that other article so we can link to the section directly.